### PR TITLE
Graphical interface for dot

### PR DESCRIPTION
**Pull request by Ghassen-Chaabouni**
## Description

This PR adds a graphical interface for dot.

To test it run `python src/dot/ui/ui.py`

gui_dot_demo

#### Added:

- `src/dot/ui/ui.py` a graphical interface for dot using `tkinter` and `customtkinter`

#### Updated:

- Path of `CACHED_PREDICTOR_PATH`

### CODE REVIEW

**Review by ajndkr — COMMENTED** [review]
looks good. please update readme with usage instructions.

**Comment by ajndkr** [comment]
@Ghassen-Chaabouni can you also resolve all dependabot alerts in this pr? https://github.com/sensity-ai/dot/security/dependabot

**Comment by Ghassen-Chaabouni** [comment]
> @Ghassen-Chaabouni can you also resolve all dependabot alerts in this pr? https://github.com/sensity-ai/dot/security/dependabot

I think it would be better to resolve them in a seperate PR

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,12 +25,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Update python version to 3.8 by @vassilispapadop in https://github.com/sensity-ai/dot/pull/28
 * Requirements changes now trigger CI by @giorgiop in https://github.com/sensity-ai/dot/pull/27
 * Fix python3.8 pip cache location in CI by @AjinkyaIndulkar in https://github.com/sensity-ai/dot/pull/29
-* Fix `--save_folder` CLI Option by @vassilispapadop and @ghassen1302 in https://github.com/sensity-ai/dot/pull/26
+* Fix `--save_folder` CLI Option by @vassilispapadop and @Ghassen-Chaabouni in https://github.com/sensity-ai/dot/pull/26
 * Add contributors list by @AjinkyaIndulkar in https://github.com/sensity-ai/dot/pull/31
 * Add Google Colab demo notebook by @AjinkyaIndulkar in https://github.com/sensity-ai/dot/pull/33
-* Speed up SimSwap's `reverse2original` by @AjinkyaIndulkar and @ghassen1302 in https://github.com/sensity-ai/dot/pull/20
+* Speed up SimSwap's `reverse2original` by @AjinkyaIndulkar and @Ghassen-Chaabouni in https://github.com/sensity-ai/dot/pull/20
```
Comment (ajndkr) [inline CHANGELOG.md:31]: @Ghassen-Chaabouni can you please update my username as well? It’s @ajndkr now

all occurrences would be nice

File: README.md [hunk 2]
```diff
@@ -119,6 +119,23 @@ pip install -e .
 
 ### Running dot
 
+#### Graphical interface
+
+To use the graphical interface of dot run
+
+```bash
+conda activate dot
+python ./src/dot/ui/ui.py
```
Comment (ajndkr) [inline README.md:128]: I suggest you update the setup.cfg to turn this into a CLI command.

for example:

```
dot-ui -h
```

File: README.md [hunk 3]
```diff
@@ -119,6 +119,23 @@ pip install -e .
...
+```bash
+conda activate dot
+python ./src/dot/ui/ui.py
+```
+
+Watch the following demo video for better understanding of the interface
+
+<p align="center">
+<img src="./assets/gui_dot_demo.gif" width="500" height="406"/>
+</p>
```
Comment (ajndkr) [inline README.md:135]: the demo video is not sufficient enough for usage instructions. please add some instructions here on how to use the available fields.

**Comment by vassilispapadop** [comment]
@Ghassen-Chaabouni this DOT-GUI seems to work fine on Apple-M2. Though, we need to change requirements file to setup the environment for M2(out of PR's scope).

Screenshot 2023-07-04 at 3 07 10 PM

**Review by ajndkr — COMMENTED** [review]
lgtm. I still think the usage instructions in the readme are not sufficient. best to explicitly mention specific instructions for users.

**Comment by Ghassen-Chaabouni** [comment]
> lgtm. I still think the usage instructions in the readme are not sufficient. best to explicitly mention specific instructions for users.

I added a usage example in the README

**Comment by vassilispapadop** [comment]
> @giorgiop @vassilispapadop okay to merge?

Lgtm

File: README.md [hunk 4]
```diff
@@ -119,6 +119,33 @@ pip install -e .
 
 ### Running dot
 
+#### Graphical interface
+
+To use the graphical interface of dot run
+
+```bash
+conda activate dot
```
Comment (giorgiop) [inline README.md:127]: can we remove the need for ALL terminal instructions for the GUI? it defeats the purpose of building a GUI for non-technical users

Comment (giorgiop) [reply README.md:127]: what I'd like to see is a guide for people to install the GUI just by download and clicks

**Review by giorgiop — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (giorgiop) [reply README.md]: @Ghassen-Chaabouni I was asking about the need for conda. is that a requirement for installation?

Comment (Ghassen-Chaabouni) [reply README.md]: We won't need it to run the executable

**Review by giorgiop — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (Ghassen-Chaabouni) [reply README.md]: If you are asking for the case without the executable, then we need it to run the gui

Comment (giorgiop) [reply README.md]: all clear, LGTM

**Pull request merged**